src/main/java/de/tum/in/www1/artemis/web/rest/RatingResource.java

```diff
@@ -0,0 +1,120 @@
+        this.ratingService = ratingService;
+        this.userService = userService;
+        this.authCheckService = authCheckService;
+        this.resultService = resultService;
+        this.participationService = participationService;
+    }
+
+    /**
+     * Return Rating referencing resultId or 404 Not Found
+     *
+     * @param resultId - Id of result that is referenced with the rating
+     * @return Rating or 404 Not Found
+     */
+    @GetMapping("/results/{resultId}/rating")
+    @PreAuthorize("hasAnyRole('USER', 'TA', 'INSTRUCTOR', 'ADMIN')")
+    public ResponseEntity<Rating> getRatingForResult(@PathVariable Long resultId) {
+        Optional<Rating> rating = this.ratingService.findRatingByResultId(resultId);
+        if (rating.isEmpty()) {
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Rating does not exist!");
+        }
+        return ResponseEntity.ok(rating.get());
+    }
+
+    /**
+     * Persist a new Rating
+     *
+     * @param resultId    - Id of the rating that should be updated
+     * @param ratingValue - Value of the updated rating
+     * @return inserted Rating
+     * @throws URISyntaxException if the Location URI syntax is incorrect
+     */
+    @PostMapping("/results/{resultId}/rating/{ratingValue}")
+    @PreAuthorize("hasAnyRole('USER', 'TA', 'INSTRUCTOR', 'ADMIN')")
+    public ResponseEntity<Rating> createRatingForResult(@PathVariable Long resultId, @PathVariable Integer ratingValue) throws URISyntaxException {
+        User user = userService.getUser();
+        Result res = resultService.findOne(resultId);
+        StudentParticipation participation = participationService.findOneStudentParticipation(res.getParticipation().getId());
+        if (!authCheckService.isOwnerOfParticipation(participation, user)) {
+            return forbidden();
+        }
+
+        Rating result = this.ratingService.saveRating(resultId, ratingValue);
+        return ResponseEntity.created(new URI("/api/results/" + result.getId() + "/rating")).body(result);
+    }
+
+    /**
+     * Update a Rating
+     *
+     * @param resultId    - Id of the rating that should be updated
+     * @param ratingValue - Value of the updated rating
+     * @return updated Rating
+     */
+    @PutMapping("/results/{resultId}/rating/{ratingValue}")
+    @PreAuthorize("hasAnyRole('USER', 'TA', 'INSTRUCTOR', 'ADMIN')")
+    public ResponseEntity<Rating> updateRatingForResult(@PathVariable Long resultId, @PathVariable Integer ratingValue) {
+        User user = userService.getUser();
+        Result res = resultService.findOne(resultId);
+        StudentParticipation participation = participationService.findOneStudentParticipation(res.getParticipation().getId());
+        if (!authCheckService.isOwnerOfParticipation(participation, user)) {
+            return forbidden();
+        }
+
+        Rating result = this.ratingService.updateRating(resultId, ratingValue);
```

Comment:
`this.` is not needed.

Reply:
The name `result` is pretty confusing here since it's actually a `Rating` and not a `Result`. Better call it `rating` or `updatedRating`.